python/xorq/expr/api.py

```diff
@@ -203,18 +204,14 @@ def read_postgres(
 
 
 @functools.cache
-def _cached_with_op(op, pretty):
-    from xorq.config import _backend_init
-
-    con = _backend_init()
-
+def _cached_with_op(op, pretty, compiler):
     expr = op.to_expr()
-    sg_expr = con.compiler.to_sqlglot(expr)
+    sg_expr = compiler.to_sqlglot(expr)
     sql = sg_expr.sql(dialect=DataFusion, pretty=pretty)
     return sql
 
 
-def to_sql(expr: ir.Expr, pretty: bool = True) -> SQLString:
+def to_sql(expr: ir.Expr, compiler=Backend.compiler, pretty: bool = True) -> SQLString:
```

Comment:
i think we should default to the `expr`'s compiler (and possibly promote `normalize_expr`'s `get_compiler` to the `api` module for reuse